test/unit/org/apache/cassandra/index/sai/cql/LuceneAnalyzerTest.java

```diff
@@ -54,6 +54,94 @@ public void testQueryAnalyzer() throws Throwable
         assertEquals(0, execute("SELECT * FROM %s WHERE val : 'query'").size());
     }
 
+    @Test
+    public void ensureQueryAnalyzerWorksForAnalyzerMatches() throws Throwable
+    {
+        createTable("CREATE TABLE %s (id text PRIMARY KEY, val text)");
+
+        // Keyword is the identity analyzer. Lowercase maps to lowercase.
+        createIndex("CREATE CUSTOM INDEX ON %s(val) USING 'StorageAttachedIndex' WITH OPTIONS = {" +
+                    "'query_analyzer': '[{\"tokenizer\":\"keyword\"},{\"filter\":\"lowercase\"}]'};");
+
+        waitForIndexQueryable();
+
+        execute("INSERT INTO %s (id, val) VALUES ('1', 'the test')");
+
+        // This query analyzer is only a lowercasing function, so a term match returns 0 rows
+        assertEquals(0, execute("SELECT * FROM %s WHERE val : 'THE'").size());
+        // The query analzyer lowercases the term, so each result in a match
+        assertEquals(1, execute("SELECT * FROM %s WHERE val : 'THE TEST'").size());
+        assertEquals(1, execute("SELECT * FROM %s WHERE val : 'THE test'").size());
+        assertEquals(1, execute("SELECT * FROM %s WHERE val : 'the test'").size());
+    }
+
+
+    // When only the query analyzer is configured, the query is analyzed (parsed) and then the analyzed result is used
+    // to determine equality based for the indexed column, which itself is not analyzed.
+    @Test
+    public void ensureQueryAnalyzerDoesNotWorkForEquality() throws Throwable
+    {
+        createTable("CREATE TABLE %s (id text PRIMARY KEY, val text)");
+
+        // Keyword is the identity analyzer. Lowercase maps to lowercase.
+        createIndex("CREATE CUSTOM INDEX ON %s(val) USING 'StorageAttachedIndex' WITH OPTIONS = {" +
+                    "'query_analyzer': '[{\"tokenizer\":\"keyword\"},{\"filter\":\"lowercase\"}]'};");
+
+        waitForIndexQueryable();
+
+        execute("INSERT INTO %s (id, val) VALUES ('1', 'the test')");
+
+        assertThatThrownBy(() -> execute("SELECT * FROM %s WHERE val = 'THE TEST'"))
```

Comment:
Without the change to the `isAnalyzed` method, this SELECT command would return the one row, which seems wrong to me.